extensions/ql-vscode/src/query-history.ts

```diff
@@ -548,12 +543,47 @@ export class QueryHistoryManager extends DisposableObject {
     );
   }
 
+  private registerToRemoteQueriesEvents() {
+    this.remoteQueriesManager.onRemoteQueryAdded(event => {
```

Comment:
Pretty sure these need to be added to the subscription for disposal. All of the `onRemoteQuery*` event handlers.

Reply:
Good catch! Sorted.